types/rudderstack-analytics.ts

```diff
@@ -0,0 +1,142 @@
+export interface IntegrationOptions {
```

Comment:
rudder-sdk-js types

Reply:
why do we need to manually define the rudderstack types, are they not offered by the NPM package?

Reply:
this issue implies they are available for import: https://github.com/rudderlabs/rudder-sdk-js/issues/1130

Reply:
They do expose some types but its not perfect. We are able to export properties such as `ApiObject`, but they do not export the whole `RudderAnalytics` class which would give us access to typing information about all the methods in this class. 

Then, in this [(1) file (index.t.ts)](https://github.com/rudderlabs/rudder-sdk-js/blob/6b1393c4505c87753f007771b21a39adf0d6cc95/src/service-worker/index.d.ts#LL68C23-L68C23), they export the `Analytics` object as we need it, except it is not actually compatible with the core `Analytics` class found [(2) here](https://github.com/rudderlabs/rudder-sdk-js/blob/production-staging/src/core/analytics.js) as (1) exports `Flush()` as a method, but the (2) Core Analytics class does not expose `Flush()` as well as (2) exposes `load()` while (1) does not expose it.  

I figured creating my own type would be helpful here. Do you have any other suggestions to go ahead with this?